modules/core/client/directives/pagination.client.directive.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+//Pagination Directive
+angular.module('core').directive('meanPagination', function () {
+    return {
+        restrict: 'AEC',
+        scope: {
+            meanPaginationData: '=meanPaginationData',
+            meanPaginationItemCount: '=meanPaginationItemCount',
+            meanPaginationUrl: '@meanPaginationUrl',
+            meanPaginationSearch: '@meanPaginationSearch'
+        },
+        controller: ['$scope', '$filter', function($scope, $filter) {
+
+            $scope.buildPager = function () {
+                $scope.pagedItems = [];
+                $scope.itemsPerPage = $scope.meanPaginationItemCount;
+                $scope.currentPage = 1;
+                $scope.figureOutItemsToDisplay();
+            };
+
+            $scope.figureOutItemsToDisplay = function () {
+                $scope.filteredItems = $filter('filter')($scope.meanPaginationData, {
+                    $: $scope.search
+                });
+                $scope.filterLength = $scope.filteredItems.length;
```

Comment:
modules/core/client/directives/pagination.client.directive.html

it is required for the angular bootstrap ui.  If you enter a search criteria it has to recalculate how many total items there are based on the filter.